pkg/binding/field_copy_test.go

```diff
@@ -0,0 +1,111 @@
+package binding
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type splitPathResult struct {
+	StartPath        []string
+	EndPath          []string
+	IsSlice          bool
+	SliceIndexString string
+}
+
+func TestNestedFieldCopy(t *testing.T) {
```

Comment:
the function under test is getValueByJSONPath, so let's reflect that in the test name.